src/lightning/__setup__.py

```diff
@@ -24,6 +26,38 @@ def _load_py_module(name: str, location: str) -> ModuleType:
 _SETUP_TOOLS = _load_py_module("setup_tools", os.path.join(_PROJECT_ROOT, ".actions", "setup_tools.py"))
 
 
+def _prepare_extras() -> Dict[str, Any]:
+    # https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras
+    # Define package extras. These are only installed if you specify them.
+    # From remote, use like `pip install pytorch-lightning[dev, docs]`
+    # From local copy of repo, use like `pip install ".[dev, docs]"`
+    req_files = [Path(p) for p in glob.glob(os.path.join(_PATH_REQUIREMENTS, "*", "*.txt"))]
+    common_args = dict(unfreeze="major" if _FREEZE_REQUIREMENTS else "all")
+    reqs = [
+        dict(
+            dir=p.parent,
+            fname=p.name,
+            extra=p.stem,
+            group=p.parent.name,
+            req=_SETUP_TOOLS.load_requirements(file_name=p.name, path_dir=p.parent, **common_args),
+        )
+        for p in req_files
+        if p.name not in ("docs.txt", "devel.txt", "base.txt")
+    ]
+
+    extras = {f"{r['group']}_{r['extra']}": r["req"] for r in reqs}
+    extra_names = list(extras.keys())
+    for extra in extra_names:
+        name = "_".join(extra.split("_")[1:])
```

Comment:
No strong preference but I prefer to split with `-`

```suggestion
        name = "-".join(extra.split("_")[1:])
```

Reply:
I suggest this because something like `pytorch_strategies` looks like a PyPI package.

Reply:
ok, but then it is elsewhere in `f"{}_{}"`